src/pages/settings/addressbook/addressbook.ts

```diff
@@ -39,13 +41,27 @@ export class AddressbookPage {
         this.isEmptyList = _.isEmpty(addressBook);
 
         let contacts: object[] = [];
+        const promises = [];
         _.each(addressBook, (contact, k: string) => {
-          contacts.push({
-            name: _.isObject(contact) ? contact.name : contact,
-            address: k,
-            email: _.isObject(contact) ? contact.email : null
-          });
+          promises.push(
+            this.getAddressOrder(k).then(value => {
+              let entry = {
+                address: contact['address'],
+                email: contact['email'],
+                name: contact['name'],
+                order: value
+              };
+
+              contacts.push(entry);
+              return Promise.resolve();
+            })
+          );
+        });
+
+        Promise.all(promises).then(() => {
+          this.addressbook = contacts;
         });
+
         this.addressbook = _.clone(contacts);
```

Comment:
this is not needed, you already assign the contacts to address book inside the "then"

Reply:
These two have been removed.